features/theme/index.tsx

```diff
@@ -145,7 +152,8 @@ export function createTheme({
                 if (next.ignoreSystemMode) {
                     Colors.setScheme(prev.scheme || defaultScheme);
                 } else {
-                    Colors.setScheme('default');
+                    const systemScheme = Appearance.getColorScheme() as 'light' | 'dark' | null;
+                    Colors.setScheme(systemScheme || 'default');
                 }
```

Comment:
This logic seems to be repeated again; please write a function to handle all 3 places.